src/bin/idl/mod.rs

```diff
@@ -20,7 +20,7 @@ use std::{
 pub fn idl(matches: &ArgMatches) {
     let files = matches.get_many::<OsString>("INPUT").unwrap();
 
-    let output = matches.get_one::<OsString>("OUTPUT").map(PathBuf::from);
+    let output = matches.get_one::<String>("OUTPUT").map(PathBuf::from);
```

Comment:
This shouldn't be `String`, that would break non-utf8 paths. If anything this should be `PathBuf`.

Reply:
If the return value is not a string, the compiler panics with the following error:
`thread 'main' panicked at 'Mismatch between definition and access of `OUTPUT`. Could not downcast to std::ffi::os_str::OsString, need to downcast to alloc::string::String
', src/bin/idl/mod.rs:23:26`

This is because the definition of the command does not specify any type, so it defaults to string:

https://github.com/hyperledger/solang/blob/e827fe52b89cb86d4db45100d6e94c29ef68a49f/src/bin/solang.rs#L104-L110

In addition, throughout the code we refer to the output as String, like here:

https://github.com/hyperledger/solang/blob/e827fe52b89cb86d4db45100d6e94c29ef68a49f/src/bin/solang.rs#L347-L353

and here:

https://github.com/hyperledger/solang/blob/e827fe52b89cb86d4db45100d6e94c29ef68a49f/src/bin/solang.rs#L518-L525

Is it time to change this everywhere?

Reply:
The right thing to do is change it everywhere. The declaration of the output arg needs either `.value_parser(ValueParser::path_buf())` or`.value_parser(ValueParser::os_string()`, depending on which makes more sense.